### PR TITLE
Add NERDTreeRefreshRoot command

### PR DESCRIPTION
**Pull request by wgfm**
Implements https://github.com/scrooloose/nerdtree/issues/727

### CODE REVIEW

**Comment by PhilRunninger** [comment]
@wgfm, thanks for the contribution. It appears you have some more work to do. While it works when NERDTree is the active window, if my active window is anything else, I get this error.

```
NERDTree: Refreshing the root node. This could take a while...
Error detected while processing function <SNR>47_refreshRoot:
line    2:
E121: Undefined variable: b:NERDTree
line    3:
E121: Undefined variable: b:NERDTree
NERDTree: Refreshing the root node. This could take a while... DONE
NERDTree: Refreshing the root node. This could take a while...
NERDTree: Refreshing the root node. This could take a while... DONE
```

Also, make sure the help file include a description of the new command.

**Comment by wgfm** [comment]
Ah, I see. It's because NERDTree is scoped to that particular buffer. I do not want to be invasive, nor do I have extensive knowledge of the codebase and vimscript. Would you have some pointers for a possible approach I could take?

I'd also understand if you don't have the time to guide me through this.

**Comment by PhilRunninger** [comment]
You've chosen a complicated program with which to learn vimscript, but that's OK. The plugin is written in a style that mimics object-oriented programming as much as a scripting language can. There is a global "object" you can access from any window. It is `g:NERDTree`, and the code is in `lib/nerdtree/nerdtree.vim`. I suggest you write a new function there that does the following:

1. Return if NERDTree is not open.
1. Get the current window.
1. Switch to the NERDTree window.
1. Refresh it.
1. Switch back to the original window.

The pieces are all there in that file's other functions and in the `s:refreshRoot()` function of `ui_glue.vim`.

Don't forget to change your `command! ... NERDTreeRefreshRoot...` to call your new function.

Give it another try. This shouldn't be too hard, and if you get stuck, I can help you out.

**Comment by wgfm** [comment]
I've added a `g:NERDTree.RefreshRoot()` function. There's a couple of things I'm not really sure about.

1. Should the `redraw` happen in `ui_glue`?
2. Should we call the new function from [`s:refreshRoot()`](https://github.com/scrooloose/nerdtree/blob/6b6cadf86ccce077212292b0585ab17e7884c81e/autoload/nerdtree/ui_glue.vim#L526) to reduce duplication?
3. The doc entry is quite short, but then again, there's not a lot going on that would require longer documentation.

**Comment by wgfm** [comment]
I have implemented the requested changes. Now, there are two unnecessary window switches happening when the root is refreshed while NERDTree is the current buffer.

File: autoload/nerdtree/ui_glue.vim [hunk 1]
```diff
@@ -525,9 +525,16 @@ endfunction
 " will be reloaded.
 function! s:refreshRoot()
     call nerdtree#echo("Refreshing the root node. This could take a while...")
+    if !g:NERDTree.IsOpen()
```
[the remark this answers is not in the document]
Comment (wgfm) [reply autoload/nerdtree/ui_glue.vim]: Good point, thanks!

**Comment by wgfm** [comment]
@PhilRunninger, thank you so much for your time. Much appreciated!

**Comment by thinker3** [comment]
I use `vim .`, but now press R can't refresh root, I have to checkout 6b6cadf : 2018-10-10 22:19

**Pull request merged**